### DISCUSSION TITLE
Suggest plugins for two factor authentication for version 3.0?

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#9523] [General]
Any one provide me the version 3.0 two way factor authentication plugin since I am using version 3.0

**Comment by sinnbeck** [comment]
Did you check the plugins page? It has this

https://filamentphp.com/plugins/jeffgreco-breezy

**Reply by mohandtharsha** [reply]
I have tried installing that But I am getting the below error,

> @php artisan package:discover --ansi

   BadMethodCallException

  Method Filament\Panel::getDomains does not exist.

  at vendor\filament\support\src\Concerns\Macroable.php:72
     68▕     {
     69▕         $macro = static::getMacro($method);
     70▕
     71▕         if ($macro === null) {
  ➜  72▕             throw new BadMethodCallException(sprintf(
     73▕                 'Method %s::%s does not exist.',
     74▕                 static::class,
     75▕                 $method,
     76▕             ));

  i   Bad Method Call: Did you mean Filament\Panel::getDomain() ?

  1   vendor\jeffgreco13\filament-breezy\routes\web.php:10
      Filament\Support\Components\Component::__call("getDomains", [])

  2   vendor\laravel\framework\src\Illuminate\Routing\Router.php:502
      Illuminate\Support\ServiceProvider::{closure}(Object(Illuminate\Routing\Router))

Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

**Reply by sinnbeck** [reply]
Are you sure you have the latest filament 3 ? The panel class should have a getDomains method

![image](https://github.com/filamentphp/filament/assets/13980140/5ebf480f-39d8-4f73-8004-f65b2ff76bc4)

**Reply by mohandtharsha** [reply]
Yes filament version is 3.0

**Reply by mohandtharsha** [reply]
getDomains() method is not there

**Reply by sinnbeck** [reply]
Can you check your vendor folder? It is in the source code
https://github.com/filamentphp/panels/blob/3.x/src/Panel.php#L32 Panel loads the HasRoutes trait

Which has the method
https://github.com/filamentphp/panels/blob/3.x/src/Panel/Concerns/HasRoutes.php#L117

And try running `composer update`

**Reply by mohandtharsha** [reply]
Panel.php has HasRoutes trait aand also HasRoutes.php contains getDomains() method  public function getDomain(): ?string
    {
        return $this->domain;
    }
